src/query/graphite/native/builtin_functions.go

```diff
@@ -242,6 +243,42 @@ func timeShift(
 	}, nil
 }
 
+
+func timeSlice(ctx *common.Context, inputPath singlePathSpec, start string, end string) (ts.SeriesList, error) {
+	tzOffsetForAbsoluteTime := time.Duration(0)
+	startTime, err := graphite.ParseTime(start, time.Now(), tzOffsetForAbsoluteTime)
+	endTime, err := graphite.ParseTime(end, time.Now(), tzOffsetForAbsoluteTime)
+
+	if (err != nil) {
+		return ts.NewSeriesList(), err
+	}
+
+	input := ts.SeriesList(inputPath)
+	output := make([]*ts.Series, input.Len())
```

Comment:
As per review on another PR, prefer to do the following instead of direct index setting:
```go
output := make([]*ts.Series, 0, input.Len())
// ... 
output = append(output, renamedSlicedSeries)
```